tests/testthat/test-reactivity.r

```diff
@@ -684,10 +757,12 @@ test_that("reactive() accepts quoted and unquoted expressions", {
   expect_true(is.function(isolate(fun())))
 
 
-  # Check that environment is correct - parent environment should be this one
+  # Check that environment is correct - parent of parent environment should be
+  # this one. Note that rlang::as_function() injects an intermediate
+  # environment.
   this_env <- environment()
   fun <- reactive(environment())
-  expect_identical(isolate(parent.env(fun())), this_env)
+  expect_identical(isolate(parent.env(parent.env(fun()))), this_env)
```

Comment:
Because of the extra layer, do you think others will also run into issues? Or is that _their problem_?

The `as_function()` idea seems small enough that we could implement it or even alter the final function environment after it is created.

Reply:
There is a remote, remote possibility that others could run into this.

I don't think we can change the way that `as_function()` works. I previously used `new_function()`, which doesn't insert another environment, but that can't handle cases where a quosure has another quosure in into it. For examples, see:
  https://gist.github.com/wch/812f7c6bb0f9c5b7d73194e3e5cbf758